src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/IOutputExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Binary;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    internal static class IOutputExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryWriteBigEndian<[Primitive] T>(this IOutput self, T value) where T : struct
+        {
+            var size = Unsafe.SizeOf<T>();
+            if (self.Buffer.Length < size)
+            {
+                self.Enlarge(Unsafe.SizeOf<T>());
```

Comment:
The `Unsafe.SizeOf` method is just `sizeof(T)` written in IL, since it's the C# compiler that complains about `sizeof(T)`, the IL is legal.